src/main/kotlin/paladin/router/services/brokers/BrokerService.kt

```
@@ -0,0 +1,225 @@
+
+        val dispatcher: MessageDispatcher = dispatchService.getDispatcher(brokerName)
+            ?: throw BrokerNotFoundException("Dispatcher not found for broker: $brokerName")
+
+        // Remove the broker from the database
+        messageBrokerRepository.deleteById(dispatcher.broker.id)
+        logger.info { "Broker Service => Broker $brokerName deleted successfully" }
+
+        // Remove the dispatcher from the dispatch service to stop routing messages generated from other services
+        dispatchService.removeDispatcher(brokerName)
+        return true
+        } catch (ex: Exception){
```

detekt (1.23.7)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)